### ISSUE TITLE
tw5.com HelloThere tiddler: link to Features

### ISSUE THREAD

**Issue by BurningTreeC** [Jermolene/TiddlyWiki5#4067]
Hello there,

I believe the Features tiddler should get a link in the HelloThere tiddler so that the features list stands out, it's a pretty good starting point for discovering features and mechanisms

**Comment by BurningTreeC** [comment]
... and I believe that the various mechanisms should show up in "Features"

**Comment by BurningTreeC** [comment]
... like PopupMechanism, InfoMechanism etc.

**Comment by Jermolene** [comment]
Hi @BurningTreeC I'm not sure why you closed this, it's a good suggestion...

**Comment by morosanuae** [comment]
@BurningTreeC, @Jermolene it's a good suggestion but the "Feature" tiddler it's not very attractive it its current form. It should be some kind of WOW! presentation not a simple list with awkward titles IMO. The first page I'm looking for when faced with a new app it's the feature/capabilities page to see if the app can fulfill my needs. I don't waste my time browsing through docs. Believe me it's the most important page off all the pages.

**Comment by BurningTreeC** [comment]
@morosanuae I think you're right, we could make screenshots of features and add similar pictures like the HelloThere tiddler shows

**Comment by morosanuae** [comment]
@BurningTreeC I don't know if we can make screenshots for all the features. Not all have a visual component. And to be honest I don't know if they will help. IMO suggestive icons/pictures should do the trick. But I think it's more than that. The features page should have an attractive design. Simple, catchy, consistent and very well structured. A new user should understand it at a glance. But I think this is just the peak of an giant iceberg that involve the design of the entire docs system.

**Comment by BurningTreeC** [comment]
I've made a "Feature Cloud" using the d3.js plugin for the HelloThere Thumbnail:

![Bildschirmfoto vom 2019-07-17 07-00-20](https://user-images.githubusercontent.com/9407182/61348708-ee9be900-a861-11e9-9438-5d89cc0f8166.png)

**Comment by morosanuae** [comment]
@BurningTreeC I was thinking of something like that for the actual features tiddler:
![screenshot](https://user-images.githubusercontent.com/5772514/61355876-e81b6a80-a87d-11e9-831c-cdc4352c324e.jpg)

**Comment by BurningTreeC** [comment]
@morosanuae looks great, can you make something like this?

**Comment by morosanuae** [comment]
@BurningTreeC I think so, but would take a while to find all the appropriate icons. I will try to create a demo for some of the features to see how it goes.

**Comment by morosanuae** [comment]
@BurningTreeC and others I've created a demo at: [http://features-demo.tiddlyspot.com](http://features-demo.tiddlyspot.com). Please tell me what you think. The list is generated automatically by adding caption, icon and summary fields to each feature tiddler.

**Comment by Jermolene** [comment]
Hi @morosanuae it's a good idea. The key, as you've already shown, is to share mockups or examples lifted from elsewhere. If I may suggest, don't spend too long looking for the icons before sharing further mockups; all that's needed is some placeholders that roughly express what you want. If we adopted this I'd probably want to make the icons be part of the core icon set, which means that they would need to match the existing metrics, and we're unlikely to find such icons off-the-shelf, they'll need to be custom drawn. But we shouldn't go to that much trouble until we get a bit further through the process.

**Comment by morosanuae** [comment]
@Jermolene

> If we adopted this I'd probably want to make the icons be part of the core icon set, which means that they would need to match the existing metrics, and we're unlikely to find such icons off-the-shelf, they'll need to be custom drawn.

Can we find someone to help us create the custom icons that we need? I 'm not an expert in graphic design.

> But we shouldn't go to that much trouble until we get a bit further through the **process.**

What would be the next step in the process?

**Comment by Jermolene** [comment]
Hi @morosanuae apologies for the confusion. My previous post was attempting to say please don't worry about getting the icons right in the first instance, just use placeholders. Focusing on the icons at this point is a mistake, because we should focus on communicating the overall idea that is being proposed.

In any event, the icons would need to match the dimensions of those in the core icon library, which were hand drawn by me. We're unlikely to find off-the-shelf icons that match our metrics, so the new icons are likely to need to be custom made. It doesn't necessarily have to be done by me, but we would have to maintain the quality of the existing icons.

**Comment by morosanuae** [comment]
@Jermolene OK I understand, thanks. In conclusion, I guess we have to wait for feedback from other users if they agree with my proposal or not. Maybe some of them will be willing to help us, with the overall design, creating the necessary icons and even with the text formulations, because I'm not a native English speaker. It should be a team work!
@BurningTreeC, @pmario and others, can you share your thoughts about this?

**Comment by twMat** [comment]
Yes, before icons we need to define the purpose: The first thing to ask is "who is the reader?" IMO it is the casual visitor who wants to see what this TinkiWinki thing is all about. He (...yeah, it is a *he*) is necessarily *somewhat* computer savvy but at what level? The greatly affects what "features" are of interest:

* minimum "tech abilities" for using TW - familiar with text editors, understand basic browser concepts, etc
* "spreadsheet type level" - understands that a "website" is a html-file, maybe knows what a *macro* is. Some appreciation to hear that TW is intrinsically a database. Etc<br>
Should the text even turn to...
* hackers - knows what hacking is about, understands "serverlessness", directly allows html/css, indirectly js, and that as such it can be a front end to other stuff, etc

(We could cater to all groups and e.g use "accordion buttons" to show features of increasing "advancedness". But the default view should reasonably show what hooks "the most" visitors.)

I also think it is important to convey that "this thing is *intended* for hacking". This is what makes TW unique. Anyone coming with expectations of an already *specialized* application, say, "Evernote" will probably be disappointed. TiddlyWiki is a lump of clay, not a pot, but it does come in a nice brick shape to begin with.

**Comment by morosanuae** [comment]
@twMat We can, for example, put the features in some basic to advanced order with gradual colors, from green (basic feature), to red (advanced/more technical features). IMO it should be just a single "Features" page aimed to all kind of visitors (even from another planet :wink:). And for that purpose all the features should be explained in a language for everyone. It can be made IMO. But I don't think it should contain absolutely all the features. Only the most important/major ones. We don't want to make the "Features" page a reference one.

**Comment by twMat** [comment]
Of the current features listed, IMO the following are relevant for the three "levels" outlined above.

**basic**
- Drag and Drop
- Encryption
- Importing Tiddlers
- Audio + ExternalImages (...more "multi media"?)
- Future Proof

**medium**
- KeyboardShortcuts
- Using SVG (- rename?)
- MathML (...how/why is this specifically mentioned
- LazyLoading (advanced?)

**advanced**
- Performance Instrumentation
- Scalability
- WebServer
- StartupActions

### Disqualified

...because they're probably *assumed* or not spontaneously thought of by a casual visitor at this point in time:

- AutoSave
- DateFormat
- Editor toolbar
- InfoPanel
- Modals
- Notifications
- AlertMechanism
- PermaLinks
- SafeMode
- Upgrading

### Lacking
What is most striking is how much that is *not* mentioned, including the very first things people should hear about like:

- tagging
- linking
- listing and various forms of "organizing"

Many feature are not direct "mechanisms" but more about what is possible. (Like "Future proof" which *is* listed)

- Super hackable/modifiable
- Works for both private and public sites
- Offline / online

BTW this OP is relevant for #3925

**Comment by morosanuae** [comment]
@twMat what you have done it's a good starting point IMO. I'll wait for the feedback from others also because I don't know if this will be adopted.

**Comment by morosanuae** [comment]
I've created a second version for my initial mockup. Now the levels (mentioned by @twMat) are shown in gradual colors.

See it here: [http://features-demo.tiddlyspot.com/#Features%20(proposal)%20v2](http://features-demo.tiddlyspot.com/#Features%20(proposal)%20v2)

**Comment by BurningTreeC** [comment]
@Jermolene any thoughts about this?

**Comment by pmario** [comment]
>See it here: http://features-demo.tiddlyspot.com/#Features%20(proposal)%20v2

I do like the concept. But the text in the "cards" is way to small. Both the heading and the text. The text is 15px ... We need to use em or rem .. so browser settings do matter.

**Comment by pmario** [comment]
The https://tiddlywiki.com landing page has changed quite a bit, and IMO some suggestions made here have been implemented too. So close. Please open new issues with a "narrower" focus.
